torchao/prototype/parq/optim/quantopt.py

```diff
@@ -106,6 +109,27 @@ def quantize_(
         quants.copy_(Q)
         return q
 
+    @torch.no_grad()
+    def torchao_quantize_(self, model: torch.nn.Module, config: AOBaseConfig):
+        """Recursively call torchao.quantize_ on model using given config."""
+        self.restore_latent_params()
+        param_set = {
+            p.data_ptr()
+            for group in self.regularized_param_groups()
+            for p in group["params"]
+        }
+
+        def inner_quantize_(model):
+            for module in model.children():
+                for param in module.parameters(recurse=False):
+                    if param.data_ptr() in param_set:
+                        quantize_(module, config)
+                        break
+
+                inner_quantize_(module)
```

Comment:
@andrewor14 I think the `quantize_(model, PARQConfig(optimizer, inner_config))` interface looks cleaner, but I see a potential issue with applying this `inner_quantize_` function. PARQ uses `param_set` to filter which modules to quantize. However, `quantize_` has its own recursive function that quantizes parameters based off `filter_fn` mapping `nn.Module` -> `bool`. I'm not sure how to reconcile the two.

Reply:
I see, yeah doesn't seem trivial to reconcile these two. Maybe this is OK then

Reply:
Another idea is:

```
filter_fn = optimizer.get_filter_fn()
quantize_(model, config, filter_fn)
```
where `optimizer.get_filter_fn` returns a function that checks against the `param_set`. Then you don't need `PARQConfig`

Reply:
Another idea is:

```
filter_fn = optimizer.get_filter_fn()
quantize_(model, config, filter_fn)
```
where `optimizer.get_filter_fn` returns a function that checks against the `param_set`. Then you don't need `PARQConfig`. What do you think?